src/pcd_db_dets.erl

```diff
@@ -0,0 +1,347 @@
+  Result :: ok |
+  {error, Reason :: term()}.
+delete_object({Owner, Bucket, Key, {_BinKey, _ContentType, _BinValue, Converter}}, _Keep, Owner) ->
+  BinKey = make_key(Bucket, Key, Converter),
+  do_operation_on_table(Owner, Key,
+                        fun(TableName) ->
+                          dets:delete(TableName, BinKey)
+                        end).
+
+delete_keep(Bucket, Key, Converter, Owner) ->
+  BinKey = make_key(Bucket, Key, Converter),
+  do_operation_on_table(Owner, Key,
+                        fun(TableName) ->
+                          dets:delete(TableName, BinKey)
+                        end).
+
+delete(Bucket, Key, Converter, Owner) ->
+  delete_keep(Bucket, Key, Converter, Owner).
+
+%% get/3
+%% <a>Retrieves an erlang term and db object from Bucket with Key,
+%% with retaining db connection usin a converter</a>
+-spec get(Bucket:: term(), Key :: term(), Converter :: atom(),
+          Owner :: atom()) -> Result when
+  Result :: {ok, Value :: term(), Object :: term()} |
+  {error, Reason :: term()}.
+get(Bucket, Key, Converter, Owner) ->
+  BinKey = make_key(Bucket, Key, Converter),
+  Result = do_operation_on_table(Owner, Key,
+                                 fun(TableName) ->
+                                   dets:lookup(TableName, BinKey)
+                                 end),
+
+  case Result of
+    [{BinKey, ContentType, ContentValue, Converter} = ObjectValue] ->
+      {ok, decode_value(ContentValue, ContentType, Converter),
+       {Owner, Bucket, Key, ObjectValue}};
+    {error, Reason} ->
+      _ = lager:error("DETS Get error:~p", [Reason]),
+      {error, Reason};
+    [] ->
+      {error, notfound}
+  end.
+
+%%fetch_type(Bucket, Key, Converter, Owner) ->
+%%  case get_riak_pid(Owner) of
+%%    Pid when is_pid(Pid) ->
+%%      BinKey = encode_key(Key, Converter),
+%%      riakc_pb_socket:fetch_type(Pid, Bucket, BinKey);
+%%    WAFIT ->
+%%      _ = lager:error("Riak connection. Reason:~p", [WAFIT]),
+%%      {error, WAFIT}
+%%  end.
+%%
+%%update_type(Bucket, Key, Operation, Owner, Converter) ->
+%%  update_type(Bucket, Key, Operation, Owner, Converter, []).
+%%
+%%update_type(Bucket, Key, Operation, Owner, Converter, Options) ->
+%%  case get_riak_pid(Owner) of
+%%    Pid when is_pid(Pid) ->
+%%      BinKey = encode_key(Key, Converter),
+%%      riakc_pb_socket:update_type(Pid, Bucket, BinKey, Operation, Options);
+%%    WAFIT ->
+%%      _ = lager:error("Riak connection cannot be set up. Reason:~p", [WAFIT]),
+%%      {error, WAFIT}
+%%  end.
+
+close() ->
+  ok.
+
+%% ====================================================================
+%% Internal functions
+%% ====================================================================
+
+encode_key(Key, none) -> term_to_binary(Key);
+encode_key(Key, _Converter) when is_binary(Key) -> Key;
+encode_key(Key, Converter) ->
+  {ok, ConvertedKey} = Converter:to_json(Key),
```

Comment:
Remove the dynamic function call on line 255. Only modules that define callbacks should make dynamic calls.